temp-legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt

```diff
@@ -215,7 +216,15 @@ private fun formatExpression(expression: String, currency: String): String {
 private fun calculate(expression: String): Double? {
     return try {
         // Keval doesn't support negative numbers, so we add a zero in front of the expression
-        val modifiedExpression = if (expression.startsWith("-")) "0$expression" else expression
+        var tempExpression = ""
+        for(exp in expression) {
+            when (exp) {
```

Comment:
Nice work! Can you use `buildString { }` so its more efficient and more idiomatic. 
```kotlin
val expression = buildString {
   for(char in expression) {
      // your code
   }
}
```